src/accelerate/data_loader.py

```diff
@@ -722,9 +721,9 @@ def prepare_data_loader(
             new_dataset,
             device=device if put_on_device and state.distributed_type != DistributedType.TPU else None,
             sampler=new_batch_sampler,
-            batch_size=getattr(dataloader, "batch_size", _PYTORCH_DATALOADER_KWARGS["batch_size"]),
+            batch_size=dataloader.batch_size,
             rng_types=rng_types,
-            generator=generator,
```

Comment:
Here we should pass the main generator of the original dataloader if there was one (see comment above).

Reply:
I remove `"generator"` from `ignore_kwargs`. It should be enough.

Reply:
Indeed, resolving comment.